src/NServiceBus.Core/Features/Feature.cs

```diff
@@ -56,7 +58,7 @@ protected void Defaults(Action<SettingsHolder> settings)
         /// <summary>
         /// Called when the features is activated.
         /// </summary>
-        protected internal abstract void Setup(FeatureConfigurationContext context);
+        protected internal abstract Task Setup(FeatureConfigurationContext context, CancellationToken cancellationToken = default);
```

Comment:
This is only required because of [a single test feature](https://github.com/Particular/NServiceBus/blob/053a9fadd4aff2ad4157aef9e1fb18ac508349fd/src/NServiceBus.AcceptanceTests/Satellites/When_a_message_is_available.cs#L54) which calls `ToTransportAddress`.

Although I could have tried changing that test, I decided to keep following the rabbit hole to see where it would lead.

The question that remains is: is there any valid _production_ use case for this method being `async`? I.e. is there a valid use case for performing I/O in this method? If not, we probably should consider changing that test and reverting this change.

Reply:
I'm 👎 to change the signature of Setup so I propose to change the test instead if possible? (since the user should know the physical address of the queue they are creating a satellite for I think either hardcoding or using `GetAwaiter().GetResult()` would be ok`)

@danielmarbach  @timbussmann My get feeling is that we had good reasons for making setup non-async, do you remember?